### PR TITLE
fix delegators config key

### PR DESCRIPTION
**Pull request by pine3ree**

### CODE REVIEW

**Comment by samsonasik** [comment]
I think the delegator factory class need to be updated as well to implements DelegatorFactoryInterface

**Comment by pine3ree** [comment]
Hello @samsonasik, I agree...

File: doc/book/cookbook/route-specific-pipeline.md [hunk 1]
```diff
@@ -62,7 +62,7 @@ use Zend\ServiceManager\DelegatorFactoryInterface;
 use Zend\ServiceManager\ServiceLocatorInterface;
 use Zend\Stratigility\MiddlewarePipe;
 
-class ApiResourcePipelineDelegatorFactory
+class ApiResourcePipelineDelegatorFactory implements DelegatorFactoryInterface
```
Comment (samsonasik) [inline doc/book/cookbook/route-specific-pipeline.md]: it is better to use `DelegatorFactoryInterface` from `ServiceManager\Factory` namespace so we use : 
```php
public function __invoke(
      ContainerInterface $container,
      $name,
      callable $callback,
      array $options = null
)
```

**Comment by pine3ree** [comment]
@samsonasik 
Yes, right. That was still the deprecated delegator factory.

**Comment by pine3ree** [comment]
@samsonasik 
I hope I didn't miss anything this time. Please verify, :-) thank You. Kind regards.

**Comment by samsonasik** [comment]
it seems ok now 👍

**Comment by weierophinney** [comment]
There's a problem with this patch: the original documentation was demonstrating the zend-servicemanager v2 API, but the changes introduced show only the v3 API. The Expressive installer, and the default configuration, supports *both*. Ideally, we should be:

- implementing `Zend\ServiceManager\DelegatorFactoryInterface`, as originally outlined, *but*
- implementing the methods in both the v2 and v3 versions of the interface.

This means more code in the examples, but will ensure that the code, if copy and pasted into an application, will work regardless of the zend-servicemanager version installed by the user.

If I can get back to this today, I'll use your changes as a starting point. If you do not see activity from me, and you have a chance to update, please do.

Thanks!

**Comment by pine3ree** [comment]
@weierophinney , sure! kind regards!

**Comment by weierophinney** [comment]
@pine3ree I'd updated the zend-form examples with a previous documentation patch, and have now pushed another patch that updates the route-specific pipeline examples. Both now demonstrate both v2 and v3 compatibility for the delegator factory examples.

**Pull request closed without merging**